final/scripts/app.js

```diff
@@ -149,37 +167,64 @@
    *
    ****************************************************************************/
 
-  // Gets a forecast for a specific city and update the card with the data
+  /*
+   * Gets a forecast for a specific city and updates the card with the data.
+   * getForecast() first checks if the weather data is in the cache. If so,
+   * then it gets that data and populates the card with the cached data.
+   * Then, getForecast() goes to the network for fresh data. If the network
+   * request goes through, then the card gets updated a second time with the
+   * freshest data.
+   */
   app.getForecast = function(key, label) {
-    var url = 'https://publicdata-weather.firebaseio.com/';
-    url += key + '.json';
+    /*
+     * IDs map to Yahoo Weather API Where On Earth Identifiers (WOEIDs).
+     * https://developer.yahoo.com/weather/documentation.html#req
+     */
+    var ids = {
```

Comment:
Instead of keeping the id's here, why not move them directly into the `<option value="2357536">Austin, TX</option>`? Thus you immediately get the ID without having to look it up.